files/package.json

```diff
@@ -5,24 +5,20 @@
   "license": "MIT",
   "author": "",
   "scripts": {
-    "prepare": "yarn workspace <%= addonName %> run prepare",
-    "start": "npm-run-all --parallel start:*",
+    "prepare": "yarn build",
+    "build": "yarn workspace <%= addonName %> run build",
+
+    "start": "concurrently 'npm:start:*' --restart-after 5000 --prefix-colors cyan,white,yellow",
     "start:addon": "yarn workspace <%= addonName %> run start",
-    "start:test-app": "yarn workspace test-app run start",
-    "lint": "npm-run-all --aggregate-output --continue-on-error --parallel \"lint:!(fix)\"",
-    "lint:addon": "yarn workspace <%= addonName %> run lint",
-    "lint:test-app": "yarn workspace test-app run lint",
-    "lint:fix": "npm-run-all --aggregate-output --continue-on-error --parallel lint:fix:*",
-    "lint:fix:addon": "yarn workspace <%= addonName %> run lint:fix",
-    "lint:fix:test-app": "yarn workspace test-app run lint:fix",
-    "test": "npm-run-all --aggregate-output --continue-on-error --parallel \"test:!(watch)\"",
-    "test:watch": "npm-run-all --aggregate-output --continue-on-error --parallel test:watch:*",
-    "test:test-app": "yarn workspace test-app run test",
-    "test:watch:test-app": "yarn workspace test-app run test:watch",
-    "test:watch:addon": "yarn workspace <%= addonName %> run start"
+    "start:test": "yarn workspace <%= testAppInfo.name.dashed %> run start",
+
+    "lint": "yarn workspaces run lint",
+    "lint:fix": "yarn workspaces run lint:fix",
+
+    "test": "yarn workspace <%= testAppInfo.name.dashed %> run test"
```

Comment:
By hardcoding `test-app` here, it would require some changes if people would want to have tests not only in that app, but maybe in a second test-app, or some unit-tests in the addon itself. 

Alternatives:
* run `yarn workspaces run test` (the addon already has a dummy `test` script to not error). But again, this does not allow for parallel execution, which is even more undesirable)
* make this use `concurrently`, and again have explicit `test:*` scripts. So people could easily add additional test scripts.

Reply:
A couple thoughts (tho, overall, I'm :shrug: ) :
 - I think this blueprint should optimize for what it provides, rather than try to cover all scenarios out of the box -- it's not too hard to observe how the script batching / concurrent running works an have authors adapt as needed
  - having too many scripts out of the box may have a false sense of complexity for folks new to ember (or addon/library development in general)
  - I think it'd be totally reasonable to add some interactivity to the generator (though I don't think ember-cli supports any of that? not sure) -- where we ask how many test apps folks want.
  - In general, I think trying to set up everything all at once at the beginning is too hard -- everyone wants different stuff -- something I'd like to see is a tool that allows augmenting existing projects. like.. adding a test-app, second addon, docs app, or whatever, and wiring it up correctly to your monorepo layout (this is obvs very complex to do well, and probably why we don't see a lot of it -- though there is an attempt at that here: https://github.com/warp-drive-engineering/engage )

Reply:
All of that also makes sense.

However I would still lean towards `yarn workspaces run test`, as 
* `test` of the addon is basically a noop, so no perf issue
* adding a second test-app would "just work"
* it is aligned with `lint`
* it does not need additional scripts, so it's just as simple

But that's also just a 60%/40% type of decision for me, so not insisting...

Reply:
>  though there is an attempt at that here: https://github.com/warp-drive-engineering/engage

TIL 🧐

Reply:
> I think it'd be totally reasonable to add some interactivity to the generator (though I don't think ember-cli supports any of that? not sure)

I *think* it's doable. Some hooks are async, like `beforeInstall`, so you can make the process wait until the user has made its selections, using e.g. the prompt features from https://github.com/ember-cli/console-ui

Reply:
> However I would still lean towards yarn workspaces run test, as

yeah, this makes a lot of sense to me, and is the same philosophy as the lint delegation.

> Some hooks are async,

yeah, I feel like last I looked at this tho, the `locals` was derived too early or something? idk -- we could probably hack it by storing stuff in module space, and ignore "locals" a bit? idk. problem for another PR :sweat_smile:

Reply:
updated